src/uu/env/src/env.rs

```diff
@@ -387,7 +387,17 @@ impl EnvAppData {
         original_args: &Vec<OsString>,
     ) -> UResult<Vec<OsString>> {
         let mut all_args: Vec<OsString> = Vec::new();
-        for arg in original_args {
+        let mut process_flags = true;
+        for (n, arg) in original_args.iter().enumerate() {
+            let arg_str = arg.to_string_lossy();
+            // Stop processing env flags once we reach the command or -- argument
+            if 0 < n && (arg == "--" || !(arg_str.starts_with('-') || arg_str.contains('='))) {
+                process_flags = false;
+            }
```

Comment:
I'm going more for readability than performance here. To make it a little faster I could also add `process_flags` to the conditional, like:
```rust
if process_flags && 0 < n && (arg == "--" || !(arg_str.starts_with('-') || arg_str.contains('='))) {
    ...
}
```
Then we don't need to call string methods on each iteration. But the conditional is already a little complicated and hard to read IMO, and performance gains seem negligible.